### PR TITLE
RRD-23: Create auth middelware

### PR DESCRIPTION
**Pull request by ignacioaristo**
# [projectCode-XXX: title-of-ticket](JIRA_url)

> Ticket code, title of the ticket and as a hyperlink, the corresponding JIRA link

## Type of change

> Exchange rate entered. The possible values ​​for this section are:
>
> - Feature (new feature of the server/application)
> - Bug Fix (problem solution in development environment)
> - Hot Fix (problem solution in production environment)
> - Environment Merge (merge branches to different environments)

## Description

> - Brief description of the changes introduced in this PR.
> - It must write what was changed, why it was changed and for what purpose.
> - It is recommended to describe it in points to facilitate understanding.

## Attachments

> Images or videos showing the behavior of what is introduced in the UI, necessary to facilitate the understanding and correction of the PR.
> In the event that what has been done cannot be visually detected (for example, work on the Backend), this section will remain empty.

## Change Procedure

> Elements to take into account when merging or reviewing the PR. For example:
> If the mergeo depends on another PR
> The priority of the mergeo of it
> What variables must be taken into account in order to review the PR

### CODE REVIEW

**Review by julianv97 — APPROVED** [review]

File: src/middlewares/auth.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export const checkToken = async (req: RequestWithFirebase, res: Response, next: NextFunction) => {
+  if (!req.headers.authorization) {
+    return res.boom.badRequest('Token is required');
+  }
+
+  return Firebase.auth()
+    .verifyIdToken(req.headers.authorization)
+    .then((decodedToken) => {
+      req.firebaseUid = decodedToken.uid;
```
Comment (martinraveglia) [inline src/middlewares/auth.ts]: `res.locals` is intended to be used this way, storing data which will be used by further middlewares/controllers
```suggestion
      res.locals.firebaseUid= decodedToken.uid;
```

Adding the role might be useful as well

File: src/middlewares/auth.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export interface RequestWithFirebase extends Request {
+  firebaseUid?: string;
+}
+
+export const checkToken = async (req: RequestWithFirebase, res: Response, next: NextFunction) => {
+  if (!req.headers.authorization) {
+    return res.boom.badRequest('Token is required');
+  }
+
+  return Firebase.auth()
```
Comment (martinraveglia) [inline src/middlewares/auth.ts]: using async...await might be better for maintainability and scalability issues

**Review by martinraveglia — CHANGES_REQUESTED** [review]

File: src/middlewares/auth.ts (hunk 1 above)
Comment (ignacioaristo) [reply src/middlewares/auth.ts]: Fixed.

File: src/middlewares/auth.ts (hunk 2 above)
Comment (ignacioaristo) [reply src/middlewares/auth.ts]: Done.

**Review by martinraveglia — APPROVED** [review]

**Pull request merged**